### PR TITLE
renew

### PR DESCRIPTION
**Pull request by jasenk2**

### CODE REVIEW

File: src/hal/drivers/mesa-hostmot2/sserial.c [hunk 1]
```diff
@@ -1344,14 +1344,14 @@ void hm2_sserial_prepare_tram_write(hostmot2_t *hm2, long period){
                                         double temp = *pin->float_pin;
                                         memcpy(&buff, &temp, sizeof(double));
                                     } else {
-                                        HM2_ERR_NO_LL("sserial write: LBP_FLOAT of bit-length %i not handled\n", conf->DataLength);
-                                        conf->DataType = 0; // only warn once, then ignore
+                                        HM2_ERR_NO_LL("sserial read: LBP_FLOAT of bit-length %i not handled\n", conf->DataLength);
```
Comment (SebKuzminsky) [inline src/hal/drivers/mesa-hostmot2/sserial.c]: Why change this from "write" to read"?  It's in the hm2_sserial_prepare_tram_write() function, i think the original message is correct.

File: .gitignore [hunk 2]
```diff
@@ -40,4 +40,6 @@ oprofile*
 *.log
 position.txt
 *.9
-*.glade.h
```
Comment (SebKuzminsky) [inline .gitignore]: Why did you remove this line?

File: src/hal/drivers/mesa-hostmot2/sserial.c [hunk 3]
```diff
@@ -1090,7 +1090,7 @@ int hm2_sserial_create_pins(hostmot2_t *hm2, hm2_sserial_remote_t *chan){
                 if (r < 0) {
                     HM2_ERR("error adding pin '%s', aborting\n", name);
                     return r;
-                }
+                };
```
Comment (SebKuzminsky) [inline src/hal/drivers/mesa-hostmot2/sserial.c]: No need for a semicolon here.

File: src/hal/drivers/mesa-hostmot2/sserial.c [hunk 4]
```diff
@@ -1534,14 +1534,12 @@ int hm2_sserial_read_pins(hm2_sserial_remote_t *chan){
                 *pin->float_pin = (double)(pin->accum - pin->offset) / pin->fullscale ;
                 break;
             case LBP_FLOAT:
-                if (conf->DataLength == sizeof(float) * 8){
-                    float temp;
-                    memcpy(&temp, &buff, sizeof(float));
-                    *pin->float_pin = temp;
+                if (conf->DataLength == sizeof(float) * 8 ){
+                                        float temp = *pin->float_pin;
+                                        memcpy(&buff, &temp, sizeof(float));
```
Comment (SebKuzminsky) [inline src/hal/drivers/mesa-hostmot2/sserial.c]: The new logic and the new indentation here are both wrong.

File: src/hal/drivers/mesa-hostmot2/sserial.c [hunk 5]
```diff
@@ -1534,14 +1534,12 @@ int hm2_sserial_read_pins(hm2_sserial_remote_t *chan){
...
                 } else if (conf->DataLength == sizeof(double) * 8){
-                    double temp;
-                    memcpy(&temp, &buff, sizeof(double));
-                    *pin->float_pin = temp;
+                                        double temp = *pin->float_pin;
+                                        memcpy(&buff, &temp, sizeof(double));
```
Comment (SebKuzminsky) [inline src/hal/drivers/mesa-hostmot2/sserial.c]: The new logic and indentation here are both wrong.

**Comment by jasenk2** [comment]
Opps, it some  old sserial.c here sorry

2017-10-02 23:47 GMT+03:00 Sebastian Kuzminsky <notifications@github.com>:

File: .gitignore (hunk 2 above)
Comment (jasenk2) [reply .gitignore]: i will try to remember, it is connected to our gui i think

Comment (SebKuzminsky) [reply .gitignore]: This is one of the reasons to write good commit messages ;-)

**Comment by jasenk2** [comment]
All other is looking OK for now

**Comment by jasenk2** [comment]
all request are done
